sections/Chapter 7 Sprint 5/virtuoso_part_2.tex

```diff
@@ -0,0 +1,115 @@
+\section{Implementing Virtuoso}
+Following the user stories laid out in section \ref{section:sprintPlanningSprint5}, we will look at implementing a database that supports RDF data, as well as SPARQL queries.
+
+The previous group responsible for \knox{} databases implemented a Apache Jena Fuseki database, as described in section \ref{currentState}.
+This implementation was, however, not suitable for \knox{}, as it needed to be restarted periodically to 'save' the data. This was due to their usage of the HDT format for storing the data\cite{knox2020}.
+
+For this reason, we decided to invest time in implementing OpenLink Virtuoso\cite{OpenLinkSoftwareVirtuoso} for the \knox{} database layer. We had already made this decision in sprint 2, as seen in chapter \ref{chapter:sprint2}.
+
+Before starting the implementation, we decided to examine whether Virtuoso should still be our choice.
+We attempted to gather feedback from the other groups in the \knox{} project, but they had very few requirements for the RDF database. All they wanted to do was store and retrieve RDF data.
+They did ask for this to be facilitated through SPARQL queries to a given endpoint, and that they could insert data in Turtle format. Our implementation of those specifications will be discussed shortly.
+
+Virtuoso supports both, which made it a good choice for \knox{} --- specification-wise. We also did some research on the performance of Virtuoso, and it seemed to do fine in various benchmarks\cite{addleseeComparingLinkedData2019}\cite{jovanovikBenchmarkingVirtuosoMighty2018}.
+
+Another potential choice could be Linked Data Fragments (LDF). However, LDF seems to use HDT, which would not solve the issue of restarting the database periodically. Therefore, Virtuoso remains our choice.
```

Comment:
Source for LDF using HDT, and that it would mean periodically restarting the database.

Reply:
Hah, I had added the source to our bib, but not into the text. The periodic restart thing is implied from last year's findings; I could not find any evidence myself.